tools/gfx-unit-test/ray-tracing-tests.cpp

```diff
@@ -0,0 +1,426 @@
+            SLANG_CHECK_ABORT(renderPipelineState != nullptr);
+
+            IShaderTable::Desc shaderTableDesc = {};
+            const char* raygenName = "rayGenShader";
+            const char* missName = "missShader";
+            shaderTableDesc.program = rayTracingProgram;
+            shaderTableDesc.hitGroupCount = 1;
+            shaderTableDesc.hitGroupNames = hitgroupNames;
+            shaderTableDesc.rayGenShaderCount = 1;
+            shaderTableDesc.rayGenShaderEntryPointNames = &raygenName;
+            shaderTableDesc.missShaderCount = 1;
+            shaderTableDesc.missShaderEntryPointNames = &missName;
+            GFX_CHECK_CALL_ABORT(device->createShaderTable(shaderTableDesc, shaderTable.writeRef()));
+        }
+
+        void renderFrame()
+        {
+            ComPtr<ICommandBuffer> renderCommandBuffer =
+                transientHeap->createCommandBuffer();
+            auto renderEncoder = renderCommandBuffer->encodeRayTracingCommands();
+            IShaderObject* rootObject = nullptr;
+            renderEncoder->bindPipeline(renderPipelineState, &rootObject);
+            auto cursor = ShaderCursor(rootObject);
+            cursor["resultTexture"].setResource(resultTextureUAV);
+            cursor["sceneBVH"].setResource(TLAS);
+            renderEncoder->dispatchRays(0, shaderTable, width, height, 1);
+            renderEncoder->endEncoding();
+            renderCommandBuffer->close();
+            queue->executeCommandBuffer(renderCommandBuffer);
+            queue->waitOnHost();
+        }
+
+        void checkTestResults()
+        {
+            ComPtr<ISlangBlob> resultBlob;
+            size_t rowPitch = 0;
+            size_t pixelSize = 0;
+            GFX_CHECK_CALL_ABORT(device->readTextureResource(
+                resultTexture, ResourceState::CopySource, resultBlob.writeRef(), &rowPitch, &pixelSize));
+
+            //writeImage("C:/Users/lucchen/Documents/test.hdr", resultBlob, width, height, rowPitch, pixelSize);
+
+            auto buffer = removePadding(resultBlob, width, height, rowPitch, pixelSize);
+            float expectedResult[16] = { 1, 1, 1, 1,
+                                         0, 0, 1, 1,
+                                         0, 1, 0, 1,
+                                         1, 0, 0, 1 };
+            auto actualData = (float*)buffer.getBuffer();
+            SLANG_CHECK(memcmp(actualData, expectedResult, sizeof(expectedResult)) == 0)
+        }
+
+        void run()
+        {
+            createRequiredResources();
+            renderFrame();
+            checkTestResults();
+        }
+    };
+
+    void simpleRayTracingTestImpl(IDevice* device, UnitTestContext* context)
+    {
+        BaseRayTracingTest test;
+        test.init(device, context);
+        test.run();
+    }
+
+    SLANG_UNIT_TEST(simpleRayTracingD3D12)
+    {
+        runTestImpl(simpleRayTracingTestImpl, unitTestContext, Slang::RenderApiFlag::D3D12);
```

Comment:
Need to check for ray-tracing support and IGNORE the test when the capability is not supported by the device.